plugins/modules/ndo_port_channel_interface.py

```diff
@@ -0,0 +1,372 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2024, Gaspard Micol (@gmicol) <gmicol@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: ndo_port_channel_interface
+short_description: Manage Port Channel Interfaces on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage Port Channel Interfaces on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.2 (NDO v4.4) and later.
+author:
+- Gaspard Micol (@gmicol)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be a Fabric Resource template.
+    type: str
+    required: true
+  name:
+    description:
+    - The name of the Port Channel Interface.
+    type: str
+    aliases: [ port_channel_interface, port_channel ]
+  uuid:
+    description:
+    - The UUID of the Port Channel Interface.
+    - This parameter can be used instead of O(port_channel_interface)
+      when an existing Virtual Port Channel Interface is updated.
+    - This parameter is required when parameter O(port_channel_interface) is updated.
+    type: str
+    aliases: [ port_channel_interface_uuid, port_channel_uuid ]
+  description:
+    description:
+    - The description of the Port Channel Interface.
+    type: str
+  node:
+    description:
+    - The node ID.
+    - This is only required when creating a new Port Channel Interface.
+    type: str
+  interfaces:
+    description:
+    - The list of used Interface IDs.
+    - Ranges of Interface IDs can be used.
+    - This is only required when creating a new Port Channel Interface.
+    type: list
+    elements: str
+    aliases: [ members ]
+  interface_policy_group_uuid:
+    description:
+    - The UUID of the Port Channel Interface Policy Group.
+    - This is only required when creating a new Port Channel Interface.
+    type: str
+    aliases: [ policy_uuid, interface_policy_uuid, interface_setting_uuid ]
+  interface_policy_group:
+    description:
+    - The Port Channel Interface Policy Group.
+    - This parameter can be used instead of O(interface_policy_group_uuid).
+    - If both parameter are used, O(interface_policy_group) will be ignored.
+    type: dict
+    suboptions:
+      name:
+        description:
+        - The name of the Interface Policy Group.
+        type: str
+      template:
+        description:
+        - The name of the template in which the Interface Policy Group has been created.
```

Comment:
should we provide the type of template here and where to configure it ( assume it is interface settings in fabric policies )?

Reply:
We should probably, added in note section